plugins/parquet-store/parquet_store.cpp

```diff
@@ -0,0 +1,722 @@
+    return f(x.row_group_size);
+  }
+
+  static const record_type& layout() noexcept {
+    static auto result = record_type{
+      {"row-group-size", count_type{}},
+    };
+    return result;
+  }
+};
+
+struct store_builder_state {
+  static constexpr const char* name = "active-parquet-store";
+  uuid id_ = {};
+  system::store_builder_actor::pointer self_ = {};
+
+  /// Actor handle of the accountant.
+  system::accountant_actor accountant_ = {};
+
+  /// Actor handle of the filesystem.
+  system::filesystem_actor fs_ = {};
+
+  /// Destination parquet file name.
+  std::filesystem::path path_ = {};
+
+  /// The table slices added to this partition.
+  std::vector<table_slice> table_slices_ = {};
+
+  /// The layout of the first record batch.
+  std::optional<type> vast_type_;
+
+  /// Number of events in this store.
+  size_t num_rows_ = {};
+
+  /// Plugin configuration.
+  configuration config_;
+};
+
+struct store_state {
+  static constexpr const char* name = "passive-parquet-store";
+  uuid id_ = {};
+  system::store_actor::pointer self_ = {};
+
+  /// Source parquet file name.
+  std::filesystem::path path_ = {};
+
+  /// Actor handle of the accountant.
+  system::accountant_actor accountant_ = {};
+
+  /// Actor handle of the filesystem.
+  system::filesystem_actor fs_ = {};
+
+  /// The table slices added to this partition.
+  /// Empty before data is retrieved from the file system.
+  std::optional<std::vector<table_slice>> table_slices_ = {};
+
+  /// Number of events in this store.
+  size_t num_rows_ = {};
+
+  /// Holds requests that did arrive while the parquet dat,a
+  /// was still being loaded from disk.
+  using request
+    = std::tuple<vast::query, caf::typed_response_promise<uint64_t>>;
+  std::vector<request> deferred_requests_ = {};
+};
+
+system::store_builder_actor::behavior_type store_builder(
+  system::store_builder_actor::stateful_pointer<store_builder_state> self,
+  system::accountant_actor accountant, system::filesystem_actor fs,
+  const uuid& id, const configuration& config);
+
+system::store_actor::behavior_type
+store(system::store_actor::stateful_pointer<store_state> self,
+      const system::accountant_actor& accountant,
+      const system::filesystem_actor& fs, const uuid& id);
+
+/// The plugin entrypoint for the parquet store plugin.
+class plugin final : public store_plugin {
+public:
+  /// Initializes the parquet plugin. This plugin has no general configuration.
+  /// @param options the plugin options. Must be empty.
+  caf::error initialize(data options) override {
+    if (caf::holds_alternative<caf::none_t>(options))
+      return caf::none;
+    if (const auto* rec = caf::get_if<record>(&options))
+      if (rec->empty())
+        return caf::none;
+    return convert(options, config_);
+    // return caf::make_error(ec::invalid_configuration, //
+    //                        "expected empty configuration under "
+    //                        "vast.plugins.parquet-store");
```

Comment:
```suggestion
    return convert(options, config_);
```

This should do the trick already, I think.